### PR TITLE
Multi Buttons - Separation of individual Button as a child block.

### PR DESCRIPTION
**Pull request by vrundakansara**
### Description
Multi Buttons - Separation of individual Button as a child block.

### Screenshots
https://share.getcloudapp.com/NQuDZgpG

### Types of changes
Improvement - Child/Parent block layout added.

### How has this been tested?
- UX wise improvement. For end-user the output remains the same. Just a better experience for editing blocks.

### Checklist:
- [X] My code is tested
- [x] My code follows accessibility standards 
- [x] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [x] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: src/blocks/buttons-child/styling.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+			"font-size" : generateCSSUnit( sizeMobile, sizeType ),
+			"line-height" : generateCSSUnit( lineHeightMobile, lineHeightType ),
+		}
+
+		tablet_selectors[" .uagb-buttons-repeater"] = {
+			"font-size" : generateCSSUnit( sizeTablet, sizeType ),
+			"line-height" : generateCSSUnit( lineHeightTablet, lineHeightType ),
+		}
+
+	if ( "desktop" == stack ) {
```
Comment (vrundakansara) [inline src/blocks/buttons-child/styling.js:96]: Is this stack option accessible in child block?

File: src/blocks/buttons/editor.scss [hunk 2]
```diff
@@ -3,11 +3,25 @@
...
+
+[data-type="uagb/buttons"] .block-editor-inner-blocks > .block-editor-block-list__layout {
+	display: flex;
+}
+
+[data-type="uagb/buttons"] .block-editor-inner-blocks > .block-editor-block-list__layout > .wp-block {
+	width: auto;
+    padding-left: 0;
+    padding-right: 0;
+    margin-left: 0;
```
Comment (vrundakansara) [inline src/blocks/buttons/editor.scss:23]: Should this be just `margin: 0;` ?

File: src/blocks/buttons/styling.js [hunk 3]
```diff
@@ -8,157 +8,85 @@ import generateCSSUnit from "../../../dist/blocks/uagb-controls/generateCSSUnit"
...
+			}
+	
+			mobile_selectors[" .block-editor-block-list__layout"] = {
+				"flex-direction": "column"
+			}
+		}
+
+		var alignment = ( align == "left" ) ? "flex-start" : ( ( align == "right" ) ? "flex-end" : "center" )
+
+		if( align == "full" ) {
```
Comment (vrundakansara) [inline src/blocks/buttons/styling.js:73]: Remove this code, because we have removed the full compatibility.

**Comment by vrundakansara** [comment]
If this file is not needed. Let's get rid of it.
`src/blocks/buttons-child/deprecated.js`

File: src/blocks/buttons/styling.js (hunk 3 above)
Comment (RajkiranBagal) [reply src/blocks/buttons/styling.js:73]: Done

File: src/blocks/buttons/editor.scss (hunk 2 above)
Comment (RajkiranBagal) [reply src/blocks/buttons/editor.scss]: Done

File: src/blocks/buttons-child/styling.js (hunk 1 above)
Comment (RajkiranBagal) [reply src/blocks/buttons-child/styling.js]: Removed

File: classes/class-uagb-block-helper.php [hunk 4]
```diff
@@ -853,10 +794,101 @@ public static function get_buttons_css( $attr, $id ) { 			// @codingStandardsIgn
...
 
+			$selectors[$wrapper . ':hover'] = array (
+				'background'   => $attr['hBackground'],
+				'border-width' => UAGB_Helper::get_css_value( $attr['borderWidth'], 'px' ),
+				'border-color' => $attr['borderHColor'],
+				'border-style' => $attr['borderStyle'],
+			);
+
+			$selectors[$wrapper . ' a.uagb-button__link'] = array (
+				'padding' => $attr['vPadding'] . 'px ' . $attr['hPadding'] . 'px',
```
[the remark this answers is not in the document]
Comment (RajkiranBagal) [reply classes/class-uagb-block-helper.php]: Done

File: classes/class-uagb-block-helper.php [hunk 5]
```diff
@@ -853,10 +794,101 @@ public static function get_buttons_css( $attr, $id ) { 			// @codingStandardsIgn
...
+		public static function get_buttons_child_selectors( $attr, $id, $childMigrate ) { 	
+
+			$wrapper = ( ! $childMigrate ) ? " .uagb-buttons-repeater-" . $id : " .uagb-buttons-repeater";
+
+			$m_selectors = array();
+			$t_selectors = array();
+
+			$selectors[$wrapper] = array (
+				'font-size'     => $attr['size'] . $attr['sizeType'],
+				'line-height'   => $attr['lineHeight'] . $attr['lineHeightType'],
```
[the remark this answers is not in the document]
Comment (RajkiranBagal) [reply classes/class-uagb-block-helper.php]: Done

File: classes/class-uagb-block-helper.php [hunk 6]
```diff
@@ -853,10 +794,101 @@ public static function get_buttons_css( $attr, $id ) { 			// @codingStandardsIgn
...
+		 */
+		public static function get_buttons_child_selectors( $attr, $id, $childMigrate ) { 	
+
+			$wrapper = ( ! $childMigrate ) ? " .uagb-buttons-repeater-" . $id : " .uagb-buttons-repeater";
+
+			$m_selectors = array();
+			$t_selectors = array();
+
+			$selectors[$wrapper] = array (
+				'font-size'     => $attr['size'] . $attr['sizeType'],
```
[the remark this answers is not in the document]
Comment (RajkiranBagal) [reply classes/class-uagb-block-helper.php]: Done

File: classes/class-uagb-block-helper.php [hunk 7]
```diff
@@ -829,15 +723,62 @@ public static function get_buttons_css( $attr, $id ) { 			// @codingStandardsIgn
...
-				$m_selectors[" .uagb-buttons__wrap"] = array (
+				$m_selectors[" .uagb-buttons-layout-wrap"] = array (
 					 "flex-direction" => "column"
 				);
+			}
+			$alignment = ( $attr['align'] == 'left' ) ? 'flex-start' : ( ( $attr['align'] == 'right' ) ? 'flex-end' : 'center' );
 
-				$m_selectors[" .uagb-button__wrapper:last-child"] = array (
-					"margin-bottom" => 0
+			if( 'full' === $attr['align'] ) {
```
[the remark this answers is not in the document]
Comment (RajkiranBagal) [reply classes/class-uagb-block-helper.php]: Done

File: classes/class-uagb-block-helper.php [hunk 8]
```diff
@@ -829,15 +723,62 @@ public static function get_buttons_css( $attr, $id ) { 			// @codingStandardsIgn
...
+					'-webkit-box-pack'=> $alignment,
+					'-ms-flex-pack' => $alignment,
+					'justify-content' => $alignment,
+					'-webkit-box-align' => $alignment,
+					'-ms-flex-align' => $alignment,
+					'align-items' => $alignment,
 				);
 			}
 
+			$selectors[' .uagb-buttons-repeater'] = array (
```
[the remark this answers is not in the document]
Comment (RajkiranBagal) [reply classes/class-uagb-block-helper.php]: Done

File: src/blocks/buttons/styling.js (hunk 3 above)
Comment (vrundakansara) [reply src/blocks/buttons/styling.js:73]: This is still pending.

Comment (RajkiranBagal) [reply src/blocks/buttons/styling.js]: Done

**Pull request merged**